From: Mykola Golub <mgolub@mirantis.com>
Date: Wed, 4 May 2016 15:25:04 +0300
Subject: rbd-mirror: avoid potential deadlock

Signed-off-by: Mykola Golub <mgolub@mirantis.com>
(cherry picked from commit 46bf727d43af4975beeb514630a99d59cfb0a406)
(cherry picked from commit 569f06f1241309a710a0570d4ede950c1dc6e5d1)
---
 src/tools/rbd_mirror/ImageSync.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/tools/rbd_mirror/ImageSync.cc b/src/tools/rbd_mirror/ImageSync.cc
index f71dfdd..2c6f5aa 100644
--- a/src/tools/rbd_mirror/ImageSync.cc
+++ b/src/tools/rbd_mirror/ImageSync.cc
@@ -166,8 +166,6 @@ void ImageSync<I>::send_copy_image() {
     return;
   }
 
-  update_progress("COPY_IMAGE");
-
   CephContext *cct = m_local_image_ctx->cct;
   ldout(cct, 20) << dendl;
 
@@ -179,6 +177,8 @@ void ImageSync<I>::send_copy_image() {
     ctx, m_progress_ctx);
   m_lock.Unlock();
 
+  update_progress("COPY_IMAGE");
+
   m_image_copy_request->send();
 }
 
